Sources/OpenGoogleSignInSDK/OpenGoogleSignInSDK.swift

```diff
@@ -148,7 +151,48 @@ public final class OpenGoogleSignIn: NSObject {
             return
         }
         
-        let task = session.dataTask(with: tokenRequest) { data, response, error in
+        makeRequest(tokenRequest) { result in
+            switch result {
+            case let .success(data):
+                do {
+                    var user = try self.decodeUser(from: data)
+                    
+                    guard let profileRequest = self.makeProfileRequest(user: user) else {
+                        completion(.success(user))
+                        return
+                    }
+                    
+                    // To obtain profile data we need to make another request.
+                    // If the request fails, we only log the error
+                    // and call completion with the `GoogleUser` object
+                    // without his profile info.
+                    self.makeRequest(profileRequest) { result in
```

Comment:
Isn't this memory leak? 🕵️

Reply:
I don't think this is a leak. Since our lib is one big singleton this cannot cause memory leak IMO